File: 0050-powx-n/0050-powx-n.cpp
class Solution {
public:
    double myPow(double x, int n) {
        double power = 1;
        long m = n;
        if(m == 0){
            return 1;
        }
        
        if(m < 0){
            x = 1/x;
            m = -m;
        }

        while(m > 0){
            if(m%2){
                power *= x;
            }
            x *= x;
            m /= 2;
        }
        return power;
    }
};

File: 0050-powx-n/0050-powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        power = 1
        if n == 0:
            return 1
        if n < 0:
            x = 1/x
            n = -n
        while n > 0:
            if n % 2:
                power *= x
            x *= x
            n //= 2
        return power
